src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,167 @@
+package duke.main;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.EventCommand;
+import duke.command.ToDoCommand;
+import duke.command.DeleteCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.FindCommand;
+import duke.exception.DukeException;
+
+/**
+ * Handles the parsing of user input commands to create corresponding Command objects.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input to create a corresponding Command object.
+     *
+     * @param userInput The user's input as a String.
+     * @return A Command object representing the parsed user input.
+     * @throws DukeException If the user input is invalid.
+     */
+    static Command parse(String userInput) throws DukeException {
+        if (userInput.equals("bye") || userInput.equals("exit")){
+            return new ExitCommand();
+        } else if (userInput.equals("list")) {
+            return new ListCommand();
+        } else if (userInput.startsWith("mark")) {
+            int taskIndex = validateMarkAndDelete(userInput.substring(5));
+            return new MarkCommand(taskIndex);
+        } else if (userInput.startsWith("unmark")) {
+            int taskIndex = validateMarkAndDelete(userInput.substring(7));
+            return new UnmarkCommand(taskIndex);
+        } else if (userInput.startsWith("todo")) {
+            String todoDescription = userInput.replace("todo ", "");
+            validateToDo(todoDescription);
+            return new ToDoCommand(todoDescription);
+        } else if (userInput.startsWith("deadline")) {
+            String[] info = validateDeadline(userInput);
+            return new DeadlineCommand(info[0], info[1]);
+        } else if (userInput.startsWith("event")) {
+            String[] info = validateEvent(userInput);
+            return new EventCommand(info[0], info[1], info[2]);
+        } else if (userInput.startsWith("delete")) {
+            int taskIndex = validateMarkAndDelete(userInput.substring(7));
+            return new DeleteCommand(taskIndex);
+        } else if (userInput.startsWith("find")) {
+            String keyword = validateFind(userInput);
+            return new FindCommand(keyword);
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
```

Comment:
A switch case would make this more readable
Would it be better to extract the phrases out and avoid hardcoding into the if-else/switch-case statements?